nifi-extension-bundles/nifi-gcp-bundle/nifi-gcp-processors/src/main/java/org/apache/nifi/processors/gcp/credentials/factory/CredentialsFactory.java

```diff
@@ -46,19 +48,35 @@
 public class CredentialsFactory {
 
     private final List<CredentialsStrategy> strategies = new ArrayList<>();
+    private final Map<AuthenticationStrategy, CredentialsStrategy> strategiesByAuthentication = new EnumMap<>(AuthenticationStrategy.class);
 
     public CredentialsFactory() {
         // Primary Credential Strategies
-        strategies.add(new ExplicitApplicationDefaultCredentialsStrategy());
-        strategies.add(new JsonFileServiceAccountCredentialsStrategy());
-        strategies.add(new JsonStringServiceAccountCredentialsStrategy());
-        strategies.add(new ComputeEngineCredentialsStrategy());
+        final CredentialsStrategy explicitApplicationDefaultCredentialsStrategy = new ExplicitApplicationDefaultCredentialsStrategy();
+        final CredentialsStrategy jsonFileServiceAccountCredentialsStrategy = new JsonFileServiceAccountCredentialsStrategy();
+        final CredentialsStrategy jsonStringServiceAccountCredentialsStrategy = new JsonStringServiceAccountCredentialsStrategy();
+        final CredentialsStrategy computeEngineCredentialsStrategy = new ComputeEngineCredentialsStrategy();
+
+        strategies.add(explicitApplicationDefaultCredentialsStrategy);
+        strategies.add(jsonFileServiceAccountCredentialsStrategy);
+        strategies.add(jsonStringServiceAccountCredentialsStrategy);
+        strategies.add(computeEngineCredentialsStrategy);
 
         // Implicit Default is the catch-all primary strategy
         strategies.add(new ImplicitApplicationDefaultCredentialsStrategy());
```

Comment:
I think `ImplicitApplicationDefaultCredentialsStrategy` should be removed as it is not used anymore. Also, `ExplicitApplicationDefaultCredentialsStrategy` could be renamed to simply `ApplicationDefaultCredentialsStrategy`.